api/src/services/extracts_v1/get_extracts.py

```diff
@@ -0,0 +1,54 @@
+import logging
+
+from pydantic import BaseModel, Field
+from sqlalchemy import select
+
+import src.adapters.db as db
+from src.constants.lookup_constants import ExtractType
+from src.db.models.extract_models import ExtractMetadata
+from src.pagination.pagination_models import PaginationParams
+from src.search.search_models import DateSearchFilter
+from src.util.file_util import pre_sign_file_location
+
+logger = logging.getLogger(__name__)
+
+
+class ExtractFilters(BaseModel):
+    extract_type: ExtractType | None = None
+    created_at: DateSearchFilter | None = None
+
+
+class ExtractListParams(BaseModel):
+    pagination: PaginationParams
+
+    filters: ExtractFilters | None = Field(default=None)
+
+
+def get_extracts(db_session: db.Session, list_params: ExtractListParams) -> list[ExtractMetadata]:
+    stmt = select(ExtractMetadata)
+
+    if list_params.filters:
+        if list_params.filters.extract_type:
+            stmt = stmt.where(ExtractMetadata.extract_type == list_params.filters.extract_type)
+
+        if list_params.filters.created_at:
+            if list_params.filters.created_at.start_date:
+                stmt = stmt.where(
+                    ExtractMetadata.created_at >= list_params.filters.created_at.start_date
+                )
+            if list_params.filters.created_at.end_date:
+                stmt = stmt.where(
+                    ExtractMetadata.created_at <= list_params.filters.created_at.end_date
+                )
+
+    # Apply pagination
+    offset = list_params.pagination.page_size * (list_params.pagination.page_offset - 1)
+    stmt = stmt.offset(offset).limit(list_params.pagination.page_size)
```

Comment:
Should we use the paginator for this? You can see it in the v0.1 version of the opportunity endpoint: https://github.com/HHS/simpler-grants-gov/blob/main/api/src/services/opportunities_v0_1/search_opportunities.py#L260

https://github.com/HHS/simpler-grants-gov/blob/main/api/src/pagination/paginator.py

Reply:
Good ida, added this